src/core/server/plugins/plugins_service.ts

```diff
@@ -253,6 +256,20 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
     this.log.debug(`Discovered ${pluginEnableStatuses.size} plugins.`);
   }
 
+  private getUnmetPluginDependencies(
+    pluginName: PluginName,
+    pluginEnableStatuses: Map<PluginName, { plugin: PluginWrapper; isEnabled: boolean }>,
+    parents: PluginName[] = []
+  ): string[] {
+    const requiredPlugins = pluginEnableStatuses.get(pluginName)?.plugin.requiredPlugins ?? [];
+    return requiredPlugins
+      .filter((dep) => !parents.includes(dep))
+      .filter(
+        (dependencyName) =>
+          !this.shouldEnablePlugin(dependencyName, pluginEnableStatuses, [...parents, pluginName])
+      );
+  }
```

Comment:
This logic is very closely tied to the `shouldEnablePlugin` function. Maybe it would make sense to have that function return a type like: `{ enabled: true } | { enabled: false, missingDependencies: string[] }` instead to reduce the duplicated logic

Reply:
Yeah I was thinking about that, I'll combine them.

Reply:
cf15a28